homeassistant/components/shelly/climate.py

```diff
@@ -0,0 +1,205 @@
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up climate device."""
+
+    if get_device_entry_gen(config_entry) == 2:
+        return
+
+    wrapper = hass.data[DOMAIN][DATA_CONFIG_ENTRY][config_entry.entry_id][BLOCK]
+    blocks = [block for block in wrapper.device.blocks if hasattr(block, "targetTemp")]
+
+    if not blocks:
+        return
+
+    async_add_entities(ShellyClimate(wrapper, block) for block in blocks)
+
+
+class ShellyClimate(ShellyBlockEntity, RestoreEntity, ClimateEntity):
+    """Representation of a KNX climate device."""
+
+    _attr_hvac_modes = [HVAC_MODE_OFF, HVAC_MODE_HEAT, HVAC_MODE_AUTO]
+    _attr_icon = "hass:thermostat"
+    _attr_max_temp = SHTRV_01_TEMPERATURE_SETTINGS["max"]
+    _attr_min_temp = SHTRV_01_TEMPERATURE_SETTINGS["min"]
+    _attr_supported_features: int = SUPPORT_TARGET_TEMPERATURE | SUPPORT_PRESET_MODE
+    _attr_target_temperature_step = SHTRV_01_TEMPERATURE_SETTINGS["step"]
+    _attr_temperature_unit = TEMP_CELSIUS
+
+    def __init__(self, wrapper: BlockDeviceWrapper, block: Block) -> None:
+        """Initialize climate."""
+        super().__init__(wrapper, block)
+
+        self.wrapper = wrapper
+        self.block = block
+        self.control_result: dict[str, Any] | None = None
+
+        self._attr_name = self.wrapper.name
+        self._attr_unique_id = f"{self.wrapper.mac}-climate"
+        self._attr_preset_modes: list[str] = wrapper.device.settings[
+            "schedule_profile_names"
+        ]
+        self._attr_preset_modes.insert(0, "None")
+
+    @property
+    def target_temperature(self) -> float | None:
+        """Set target temperature."""
+        return cast(float, self.block.targetTemp)
+
+    @property
+    def current_temperature(self) -> float | None:
+        """Set current temperature."""
+        return cast(float, self.block.temp)
+
+    # @property
+    # def is_available(self) -> bool:
+    #     """Device a vailability."""
+    #     return not cast(bool, self.block.valveError)
+
+    @property
+    def hvac_mode(self) -> str:
+        """HVAC current mode."""
+        if not hasattr(self.block, "mode") or self._check_is_off():
+            return HVAC_MODE_OFF
+
+        if cast(int, self.block.mode) != 0:
+            return HVAC_MODE_AUTO
+        return HVAC_MODE_HEAT
+
+    @property
+    def preset_mode(self) -> str | None:
+        """Preset current mode."""
+        if not hasattr(self.block, "mode"):
+            return None
+        return self._attr_preset_modes[cast(int, self.block.mode)]
+
+    @property
+    def hvac_action(self) -> str | None:
+        """HVAC current action."""
+        if self._check_is_off():
+            return CURRENT_HVAC_OFF
+        return CURRENT_HVAC_HEAT
+
+    def _check_is_off(self) -> bool:
+        """Return if valve is off or on."""
+        return bool(
+            not self.target_temperature
```

Comment:
```suggestion
            self.target_temperature is None
```
This will return off if target temperature is 0